Ve.Metrics.StatsDClient.CastleWindsor/StatsDTimingInterceptor.cs

```diff
@@ -28,10 +30,17 @@ protected override void Invoke(IInvocation invocation, StatsDTiming attr)
 
             if (methodBase.IsGenericMethod)
             {
+                var generic = new List<string>();
                 var arguments = methodBase.GetGenericArguments();
-                var generic = arguments[0].Name;
 
-                name = name.Replace("{generic}", generic.ToLowerInvariant());
+                while (arguments.Any())
+                {
+                    var argument = arguments.First();
+                    generic.Add(argument.Name.ToLowerInvariant());
+                    arguments = argument.GetGenericArguments();

```

Comment:
can you wrap some of this in a method to make it more obvious what it's doing?

Just checking i've read this right, it's searching down through the sets until it hits a concrete class?